File: source-connector/src/main/java/sourceconnector/config/ConnectConfig.java
package sourceconnector.config;

import org.springframework.boot.context.properties.ConfigurationProperties;

@ConfigurationProperties(prefix = "connect")
public record ConnectConfig(
  int workerCount,
  int taskCount
) {
  public ConnectConfig {
    if (workerCount < 1) throw new IllegalArgumentException("workerCount must be >= 1");
    if (taskCount < workerCount) throw new IllegalArgumentException("taskCount must be >= workerCount");
  }
}

File: source-connector/src/main/kotlin/sourceconnector/config/ConnectConfig.kt
package sourceconnector.config

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "connect")
data class ConnectConfig(
  val workerCount: Int,
  val taskCount: Int
) {
  init {
    require(workerCount >= 1) { "workerCount must be >= 1" }
    require(taskCount >= workerCount) { "taskCount must be >= workerCount" }
  }
}
